src/Computation/Values/FloatPower.cs

```diff
@@ -17,30 +17,28 @@ private FloatPower(IValue left, IValue right)
         _right = right;
     }
 
-    public override FPExpr AsFloat(IContext context)
+    public override FPExpr AsFloat(ISolver solver)
     {
         throw new UnsupportedSymbolicArithmeticException();
     }
 
-    public RealExpr AsReal(IContext context)
+    public RealExpr AsReal(ISolver solver)
     {
-        ArithExpr ToArithExpr(IValue value) =>
-            value switch
-            {
-                IRealValue r => r.AsReal(context),
-                _ => context.CreateExpr(c =>
-                {
-                    using var flt = value.AsFloat(context);
-                    return c.MkFPToReal(flt);
-                })
-            };
-
-        return context.CreateExpr(c =>
-        {
-            using var left = ToArithExpr(_left);
-            using var right = ToArithExpr(_right);
-            return (RealExpr) c.MkPower(left, right);
-        });
+        using var left = _left is IRealValue l
```

Comment:
I also think this was better as a local function too in order to avoid this duplication.